aea/skills/base.py

```diff
@@ -60,13 +61,13 @@
 class SkillContext:
     """This class implements the context of a skill."""
 
-    def __init__(self, agent_context: AgentContext):
+    def __init__(self, agent_context: Optional[AgentContext] = None):
         """
         Initialize a skill context.
 
         :param agent_context: the agent's context
         """
-        self._agent_context = agent_context
+        self._agent_context = cast(AgentContext, agent_context)
```

Comment:
makes no sense to me that first you declare it optional and then you cast it to non-optional...

Reply:
ok